src/template-tags/tickets.php

```diff
@@ -472,193 +483,202 @@ function tribe_tickets_get_ticket_stock_message( Tribe__Tickets__Ticket_Object $
+	 * in the correct directories for Tribe Tickets templates
+	 *
+	 * @param string      $slug The Base template name
+	 * @param null|string $name (optional) if set will try to include `{$slug}-{$name}.php` file
+	 * @param array       $data (optional) array of vars to inject into the template part
+	 * @param boolean     $echo (optional) Allows the user to print or return the template
+	 *
+	 * @return string|void It will depend if it's echoing or not
+	 **@uses Tribe__Tickets__Templates::get_template_hierarchy
+	 *
+	 */
+	function tribe_tickets_get_template_part( $slug, $name = null, array $data = null, $echo = true ) {
 
 		/**
-		 * A more Specific Filter that will include the template name
+		 * Fires an Action before echoing the Template
 		 *
-		 * @param string $file     Complete path to include the PHP File
-		 * @param string $slug     Slug for this template
-		 * @param string $name     Template name
-		 * @param array  $data     The Data that will be used on this template
+		 * @param string $slug Slug for this template
+		 * @param string $name Template name
+		 * @param array  $data The Data that will be used on this template
 		 */
-		$file = apply_filters( "tribe_tickets_get_template_part_path_{$template}", $file, $slug, $name, $data );
+		do_action( 'tribe_tickets_pre_get_template_part', $slug, $name, $data );
 
-		if ( ! file_exists( $file ) ) {
-			continue;
+		// Setup possible parts
+		$templates = array();
+		if ( isset( $name ) ) {
+			$templates[] = $slug . '-' . $name . '.php';
 		}
+		$templates[] = $slug . '.php';
 
-		ob_start();
 		/**
-		 * Fires an Action before including the template file
+		 * Allow users to filter which templates can be included
 		 *
 		 * @param string $template The Template file, which is a relative path from the Folder we are dealing with
-		 * @param string $file     Complete path to include the PHP File
 		 * @param string $slug     Slug for this template
 		 * @param string $name     Template name
 		 * @param array  $data     The Data that will be used on this template
 		 */
-		do_action( 'tribe_tickets_before_get_template_part', $template, $file, $slug, $name, $data );
-		include( $file );
+		$templates = apply_filters( 'tribe_tickets_get_template_part_templates', $templates, $slug, $name, $data );
 
-		/**
-		 * Fires an Action After including the template file
-		 * @param string $template The Template file, which is a relative path from the Folder we are dealing with
-		 * @param string $file     Complete path to include the PHP File
-		 * @param string $slug     Slug for this template
-		 * @param string $name     Template name
-		 * @param array  $data     The Data that will be used on this template
-		 */
-		do_action( 'tribe_tickets_after_get_template_part', $template, $file, $slug, $name, $data );
-		$html = ob_get_clean();
+		// Make any provided variables available in the template's symbol table
+		if ( is_array( $data ) ) {
+			extract( $data );
```

Comment:
:no_entry_sign: Extract is the devil. Use separate assignments or access the array directly instead.

<sub>hash: 819afd6ea184d45e419651a8d445553e</sub>